megamek/src/megamek/common/verifier/TestInfantry.java

```diff
@@ -197,52 +208,64 @@ public static int maxSecondaryWeapons(Infantry inf) {
      * 
      * @param movementMode  The platoon's movement mode
      * @param alt           True indicates that VTOL is microlite and INF_UMU is motorized.
+     * @param mount         The mount if the unit is beast-mounted, otherwise null.
      * @return              The maximum size of a squad.
      */
-    public static int maxSquadSize(EntityMovementMode movementMode, boolean alt) {
-        switch (movementMode) {
-            case HOVER:
-            case SUBMARINE:
-                return 5;
-            case WHEELED:
-                return 6;
-            case TRACKED:
-                return 7;
-            case INF_UMU:
-                 return alt? 6 : 10;
-            case VTOL:
-                return alt? 2 : 4;
-            default:
-                return 10;
+    public static int maxSquadSize(EntityMovementMode movementMode, boolean alt, @Nullable InfantryMount mount) {
+        if (mount == null) {
+            switch (movementMode) {
+                case HOVER:
+                case SUBMARINE:
+                    return 5;
+                case WHEELED:
+                    return 6;
+                case TRACKED:
+                    return 7;
+                case INF_UMU:
+                    return alt ? 6 : 10;
+                case VTOL:
+                    return alt ? 2 : 4;
+                default:
+                    return 10;
+            }
+        } else if (mount.getSize().troopsPerCreature == 1) {
+            return 10; // use foot infantry limit
```

Comment:
Can we define some of these magic numbers? I'm seeing at least three 10s here.

Reply:
Those are the maximum squad size defined by the rules (TM, p. 145, TO: AU&E, p. 136, 153). Each motive type has its own entry in the table. The fact that some of them are the same doesn't make them a universal value.